LWCOSC/force-app/main/default/lwc/purchaseOrderHistory/purchaseOrderHistory.js

```diff
@@ -0,0 +1,61 @@
+import { LightningElement, wire, api } from 'lwc';
+
+import PO_STATUS_FIELD from '@salesforce/schema/Purchase_Order__c.Order_Status__c';
+import PO_AMOUNT_FIELD from '@salesforce/schema/Purchase_Order__c.Order_Amount__c';
+
+import Id from '@salesforce/user/Id';
+import getPurchaseOrders from '@salesforce/apex/PurchaseOrderController.getPurchaseOrders';
+
+
+const PURCHASEORDERCOL = [
+    { label: 'PO Id', fieldName: 'Id', type: 'text' },
+    { label: 'Status', fieldName: PO_STATUS_FIELD.fieldApiName, type: 'text' },
+    { label: 'Order Total', fieldName: PO_AMOUNT_FIELD.fieldApiName, type: 'currency'}
+]; 
+
+export default class PurchaseOrderHistory extends LightningElement {
+    userId = Id;
+	@api
+    purchaseOrderDataItems;
+	@api
+	purchaseOrderColumns = PURCHASEORDERCOL;
+
+	purchaseOrderFlag = false;
+	productListFlag = false;
+
+    @wire(getPurchaseOrders, {userId: '$userId'})
+	purchaseOrderData({error, data}){
+		if(data){
+			this.purchaseOrderDataItems = data;
+			this.error = undefined;
+			if(this.purchaseOrderDataItems.length === 0){
+				this.purchaseOrderFlag = false;
+				this.productListFlag = true;
+				
+			} 
+			else{
+				this.purchaseOrderFlag = true;
+				this.productListFlag = false;
+			}
```

Comment:
this could be written as 

    const isPurchaseOrderEmpty = this.purchaseOrderDataItems.length;
    this.purchaseOrderFlag = !isPurchaseOrderEmpty;
    this.productListFlag = isPurchaseOrderEmpty

Reply:
Point Noted, As discussion No action require